spec/controllers/nodes_controller_spec.rb

```diff
@@ -116,4 +116,134 @@
       end
     end
   end
+
+  # rubocop:disable RSpec/AnyInstance
+  # rubocop:disable RSpec/ExampleLength
+  # rubocop:disable RSpec/NestedGroups
+  describe "POST /nodes/:node_id/assign" do
+    let(:salt) { Velum::Salt }
+    let(:net_httpok_instance) { Net::HTTPOK.new(1, 200, "OK") }
+    let(:net_httpservererror_instance) { Net::HTTPServerError.new(1, 500, "ServerError") }
+
+    context "HTML rendering" do
+      context "when the minion exists" do
+        before do
+          sign_in user
+          Minion.create! [{ hostname: "master" }]
+        end
+
+        it "assigns the master role" do
+          allow(salt).to receive(:orchestrate).and_return(net_httpok_instance, {})
+          allow(net_httpok_instance).to receive(:code).and_return("200")
+          allow_any_instance_of(Minion).to receive(:assign_role).with("master").and_return(true)
+          post :assign, node_id: Minion.find_by(hostname: "master").id, role: :master
+          expect(response.redirect_url).to eq "http://test.host/nodes"
+        end
+
+        # rubocop:disable RSpec/MessageChain
```

Comment:
I think (I'm not sure) that you have to enable it again when you are done or else it stays disabled.